src/CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageTransportBindingElement.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net;
+using CoreWCF.Configuration;
+using CoreWCF.Queue.Common;
+using CoreWCF.Queue.Common.Configuration;
+using Azure.Storage.Queues;
+
+namespace CoreWCF.Channels
+{
+    public class AzureQueueStorageTransportBindingElement : QueueBaseTransportBindingElement
+    {
+        /// <summary>
+        /// Creates a new instance of the AzureQueueStorageTransportBindingElement Class using the default protocol.
+        /// </summary>
+        public AzureQueueStorageTransportBindingElement()
+        {
+        }
+
+        public override BindingElement Clone()
+        {
+            return new AzureQueueStorageTransportBindingElement();
+        }
+
+
+        public override T GetProperty<T>(BindingContext context)
+        {
+            if (context == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(context));
+            }
+
+            if (typeof(T) == typeof(ISecurityCapabilities))
+            {
+                return null;
+            }
+
+            return base.GetProperty<T>(context);
+        }
+
+        public override QueueTransportPump BuildQueueTransportPump(BindingContext context)
+        {
+            IQueueTransport queueTransport = CreateMyQueueTransport(context);
+            return QueueTransportPump.CreateDefaultPump(queueTransport);
+        }
+
+        private IQueueTransport CreateMyQueueTransport(BindingContext context)
+        {
+            var serviceDispatcher = context.BindingParameters.Find<IServiceDispatcher>();
+            var serviceProvider = context.BindingParameters.Find<IServiceProvider>();
+            return new AzureQueueStorageQueueTransport(serviceDispatcher, serviceProvider);
+        }
+
+        /// <summary>
+        /// Gets the scheme used by the binding, soap.amqp
+        /// </summary>
+        public override string Scheme
+        {
+            get { return CurrentVersion.Scheme; }
+        }
+
+        /// <summary>
+        /// The largest receivable encoded message
+        /// </summary>
+        public override long MaxReceivedMessageSize { get; set; }
+
```

Comment:
I'm not seeing a default value being set which means it will have a value of 0. The default is generally 64k. I think I remember you saying there was a limit of 512MB in the client PR. This should do range checks on the setter to make sure it's not set to zero, and not set too large, as well as having a default value set in the constructor.

Reply:
I just remembered the 512MB limit was for the RabbitMQ implementation for AMQP. So this should be set to the ASQ limit of 64K and have checks to make sure it's not higher than that.

Reply:
It was getting set eventually in AQbindingElement, but it makes more sense to set it here and add validation.